x/treasury/internal/keeper/querier.go

```diff
@@ -45,6 +47,9 @@ func queryIndicators(ctx sdk.Context, keeper Keeper) ([]byte, error) {
 	taxRewards := sdk.NewDecCoinsFromCoins(keeper.PeekEpochTaxProceeds(ctx)...)
 	TR := keeper.alignCoins(ctx, taxRewards, core.MicroSDRDenom)
 
+	// Subtract remnant blocks of past chain
+	ctx = ctx.WithBlockHeight(ctx.BlockHeight() - (keeper.GetCumulatedHeight(ctx) % core.BlocksPerWeek))
```

Comment:
The mod operator and why the ctx construct tolerates a negative value deserves a bit more explanation. Perhaps: 

// The BlockHeight variable of the current context could be set to negative, in which case the querier fetches indicators from epochs corresponding to negative block heights (from a previous chain version)